project.go

```diff
@@ -43,12 +43,32 @@ func findProjectRoot(from string) (string, error) {
 type Project struct {
 	// AbsRoot is the absolute path to the root directory of the project.
 	AbsRoot string
+	// ResolvedAbsRoot is the resolved absolute path to the root directory of the project.
+	// If AbsRoot is not a symlink, then ResolvedAbsRoot should equal AbsRoot.
+	ResolvedAbsRoot string
 	// ImportRoot is the import path of the project's root directory.
 	ImportRoot gps.ProjectRoot
 	Manifest   *Manifest
 	Lock       *Lock // Optional
 }
 
+// SetRoot set the project AbsRoot. If the root is a symlink, it attempts to resolve it.
+func (p *Project) SetRoot(root string) error {
+	p.AbsRoot = root
+	p.ResolvedAbsRoot = root
+
+	// if p.AbsRoot is a symlink, set p.ResolvedAbsRoot to the resolved path.
+	if sym, err := fs.IsSymlink(p.AbsRoot); err != nil {
```

Comment:
Probably don't need to check this; `filepath.EvalSymlinks()` will just return the input if it's not a symlink, which is the desired behavior anyway.